crates/biome_js_analyze/src/semantic_analyzers/nursery/use_number_properties.rs

```diff
@@ -0,0 +1,192 @@
+use biome_diagnostics::Applicability;
+use biome_js_factory::make;
+use biome_js_syntax::{global_identifier, AnyJsExpression, JsSyntaxKind, T};
+use biome_rowan::{AstNode, BatchMutationExt};
+
+declare_rule! {
+    /// Use `Number` properties instead of global ones.
+    ///
+    /// ECMAScript 2015 moved globals onto the `Number` constructor for consistency and to slightly improve them.
+    ///
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// parseInt("1"); // true
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// parseFloat("1.1"); // true
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// NaN; // true
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// Infinity; // true
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// -Infinity; // true
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// Number.parseInt("1"); // false
+    /// ```
+    ///
+    /// ```js
+    /// Number.parseFloat("1.1"); // false
+    /// ```
+    ///
+    /// ```js
+    /// Number.NaN; // false
+    /// ```
+    ///
+    /// ```js
+    /// Number.POSITIVE_INFINITY; // false
+    /// ```
+    ///
+    /// ```js
+    /// Number.NEGATIVE_INFINITY; // false
+    /// ```
+    ///
+    pub(crate) UseNumberProperties {
+        version: "next",
+        name: "useNumberProperties",
+        recommended: true,
+        fix_kind: FixKind::Unsafe,
+    }
+}
+
+impl Rule for UseNumberProperties {
+    type Query = Semantic<AnyJsExpression>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let model = ctx.model();
+        let (reference, name) = global_identifier(node)?;
+        let names = ["parseInt", "parseFloat", "NaN", "Infinity"];
+        if !names.contains(&name.text()) {
+            return None;
+        }
+        model.binding(&reference).is_none().then_some(())
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                node.range(),
+                markup! {
+                    "Use the "<Emphasis>"Number"</Emphasis>" properties instead of the global ones."
+                },
+            )
+            .note(markup! {
+                "ECMAScript 2015 moved globals onto the "<Emphasis>"Number"</Emphasis>" constructor for consistency and to slightly improve them."
+            }),
+        )
+    }
+
+    fn action(ctx: &RuleContext<Self>, _: &Self::State) -> Option<JsRuleAction> {
+        let node = ctx.query();
+        let mut mutation = ctx.root().begin();
+        let (old, new) = match node {
+            AnyJsExpression::JsIdentifierExpression(expression) => {
+                let name = expression.name().ok()?.text();
```

Comment:
[After defining the global const variable](https://github.com/biomejs/biome/pull/1314/files#r1435639838), we can simplify the check here like: 
```rs
if !GLOBAL_NUMBER_IDENTS.contains(&name) {
	return None;
}
```

Reply:
fix: 99e0270